### PR TITLE
Add React hooks for customization (part 4)

### PR DESCRIPTION
**Pull request by compulim**
> Related to #2539.

Please merge #2540, #2541 and #2542 before reviewing this one.

> Note to myself: check test fail if it is related to missing Adaptive Cards Host Config. Also, add a test for customization the host config.

## Changelog Entry

-  Resolves [#2539](https://github.com/Microsoft/BotFramework-WebChat/issues/2539), added React hooks for customization, by [@compulim](https://github.com/compulim) and [@corinagum](https://github.com/corinagum), in the following PRs:
   -  PR [#2540](https://github.com/microsoft/BotFramework-WebChat/pull/2540): `useActivities`, `useReferenceGrammarID`, `useSendBoxDictationStarted`
   -  PR [#2541](https://github.com/microsoft/BotFramework-WebChat/pull/2541): `useStyleOptions`, `useStyleSet`
   -  PR [#2542](https://github.com/microsoft/BotFramework-WebChat/pull/2542): `useLanguage`, `useLocalize`
   -  PR [#2543](https://github.com/microsoft/BotFramework-WebChat/pull/2543): `useAdaptiveCardsHostConfig`, `useAdaptiveCardsPackage`, `useRenderMarkdownAsHTML`
- Bring your own Adaptive Cards package by specifying `adaptiveCardsPackage` prop, by [@compulim](https://github.com/compulim) in PR [#2543](https://github.com/microsoft/BotFramework-WebChat/pull/2543)

## Description

Add `useAdaptiveCardsHostConfig`, `useAdaptiveCardsPackage` and `useRenderMarkdownAsHTML`.

## Specific Changes

- Added the following hooks:
   - `useAdaptiveCardsHostConfig`
   - `useAdaptiveCardsPackage`
   - `useRenderMarkdownAsHTML`
- Updated components to use aforementioned hooks
- Added an entry prop `adaptiveCardsPackage` for easily specifying custom Adaptive Cards package

---

-  [x] Testing Added
   - [x] Adds a test custom Adaptive Cards package
   - [x] Adds a test for customizing Adaptive Cards host config

### CODE REVIEW

File: packages/component/src/Activity/StackedLayout.js [hunk 1]
```diff
@@ -88,6 +88,14 @@ const StackedLayout = ({ activity, avatarInitials, children, timestampClassName
   const [{ botAvatarInitials, bubbleNubSize, bubbleFromUserNubSize, userAvatarInitials }] = useStyleOptions();
   const [{ stackedLayout: stackedLayoutStyleSet }] = useStyleSet();
 
+  const botAriaLabel = useLocalize('Bot said something', avatarInitials, plainText);
+  const userAriaLabel = useLocalize('User said something', avatarInitials, plainText);
+  const botRoleLabel = useLocalize('BotSent');
```
Comment (tdurnford) [inline packages/component/src/Activity/StackedLayout.js]: `botRoleLabel` is already defined on line 115

File: packages/bundle/src/__tests__/createDirectLine.spec.js [hunk 2]
```diff
@@ -8,6 +8,9 @@ jest.mock('botframework-directlinejs', () => {
   return { DirectLine };
 });
 
+// We did not explicitly include React in package.json, thus, it is loading React 0.14.9.
```
Comment (tdurnford) [inline packages/bundle/src/__tests__/createDirectLine.spec.js]: Are we fixing this?

File: packages/bundle/src/adaptiveCards/hooks/useAdaptiveCardsHostConfig.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { useContext } from 'react';
+
+import AdaptiveCardsContext from '../AdaptiveCardsContext';
+
+export default function useAdaptiveCardsHostConfig() {
```
Comment (tdurnford) [inline packages/bundle/src/adaptiveCards/hooks/useAdaptiveCardsHostConfig.js]: Tests?

File: packages/bundle/src/adaptiveCards/hooks/useAdaptiveCardsPackage.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import { useContext } from 'react';
+
+import AdaptiveCardsContext from '../AdaptiveCardsContext';
```
Comment (tdurnford) [inline packages/bundle/src/adaptiveCards/hooks/useAdaptiveCardsPackage.js]: Tests?

File: samples/01.a.getting-started-full-bundle/index.html [hunk 5]
```diff
@@ -7,7 +7,8 @@
       This CDN points to the latest official release of Web Chat. If you need to test against Web Chat's latest bits, please refer to pointing to Web Chat's MyGet feed:
       https://github.com/microsoft/BotFramework-WebChat#how-to-test-with-web-chats-latest-bits
     -->
-    <script src="https://cdn.botframework.com/botframework-webchat/latest/webchat.js"></script>
+    <!-- <script src="https://cdn.botframework.com/botframework-webchat/latest/webchat.js"></script> -->
```
Comment (tdurnford) [inline samples/01.a.getting-started-full-bundle/index.html]:
```suggestion
    <script src="https://cdn.botframework.com/botframework-webchat/latest/webchat.js"></script>
```

File: samples/01.a.getting-started-full-bundle/index.html [hunk 6]
```diff
@@ -7,7 +7,8 @@
...
+    <script src="/webchat.js"></script>
```
Comment (tdurnford) [inline samples/01.a.getting-started-full-bundle/index.html]:
```suggestion
```

File: packages/component/src/Activity/StackedLayout.js (hunk 1 above)
Comment (compulim) [reply packages/component/src/Activity/StackedLayout.js]: Fixed, bad merge.

File: samples/01.a.getting-started-full-bundle/index.html (hunk 5 above)
Comment (compulim) [reply samples/01.a.getting-started-full-bundle/index.html]: Fixed.

File: samples/01.a.getting-started-full-bundle/index.html (hunk 6 above)
Comment (compulim) [reply samples/01.a.getting-started-full-bundle/index.html]: Fixed.

File: packages/bundle/src/__tests__/createDirectLine.spec.js (hunk 2 above)
Comment (compulim) [reply packages/bundle/src/__tests__/createDirectLine.spec.js]: Fixed.

File: packages/bundle/src/adaptiveCards/hooks/useAdaptiveCardsHostConfig.js (hunk 3 above)
Comment (compulim) [reply packages/bundle/src/adaptiveCards/hooks/useAdaptiveCardsHostConfig.js]: Added.

File: packages/bundle/src/adaptiveCards/hooks/useAdaptiveCardsPackage.js (hunk 4 above)
Comment (compulim) [reply packages/bundle/src/adaptiveCards/hooks/useAdaptiveCardsPackage.js]: Added.

File: packages/bundle/src/__tests__/createDirectLine.spec.js (hunk 2 above)
Comment (tdurnford) [reply packages/bundle/src/__tests__/createDirectLine.spec.js]: Should we take this comment out then?

**Pull request merged**